src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -240,11 +249,26 @@ private void storeInPluginContext(List<ImageConfiguration> resolvedImages, Strin
     private DockerAccess createDockerAccess(String minimalVersion) throws MojoExecutionException, MojoFailureException {
         DockerAccess access = null;
         if (isDockerAccessRequired()) {
-            String dockerUrl = EnvUtil.extractUrl(dockerHost);
+            boolean isDefaults = machine == null;
+            if (isDefaults) {
+                // create instance to resolve system defines
+                // docker.machine.name and docker.machine.autoCreate
+                machine = new MachineConfiguration();
+            }
+            isDefaults &= machine.resolveDefines(new PluginParameterExpressionEvaluator(session, execution), getLog());
```

Comment:
I was unable to get injection framework to provide default values for embedded instances. I'd love to be shown how to do this correctly. 

Chas

> On Jun 26, 2016, at 6:31 PM, Roland Huß <notifications@github.com> wrote:
> 
> In src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java:
> 
> > @@ -240,11 +249,26 @@ private void storeInPluginContext(List<ImageConfiguration> resolvedImages, Strin
> >      private DockerAccess createDockerAccess(String minimalVersion) throws MojoExecutionException, MojoFailureException {
> >          DockerAccess access = null;
> >          if (isDockerAccessRequired()) {
> > -            String dockerUrl = EnvUtil.extractUrl(dockerHost);
> > +            boolean isDefaults = machine == null;
> > +            if (isDefaults) {
> > +                // create instance to resolve system defines
> > +                // docker.machine.name and docker.machine.autoCreate
> > +                machine = new MachineConfiguration();
> > +            }
> > +            isDefaults &= machine.resolveDefines(new PluginParameterExpressionEvaluator(session, execution), getLog());
> Why is is needed to do an extras property resolve step here ?
> 
> For all other configurations we simply rely on Maven to have the properties resolved before injecting it into the Plugin via @Parameter. Using the same scheme here would make things much simpler.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>